src/intellisense/languageServer.node.ts

```diff
@@ -166,33 +199,24 @@ export class LanguageServer implements Disposable {
                 }
             };
 
-            languageClient = new LanguageClient('notebook-intellisense', serverOptions, clientOptions);
+            const client = new LanguageClient('notebook-intellisense', serverOptions, clientOptions);
 
             // Before starting do a little hack to prevent the pylance double command registration (working with Jake to have an option to skip commands)
             /* eslint-disable @typescript-eslint/no-explicit-any */
-            const features: (StaticFeature | DynamicFeature<any>)[] = (languageClient as unknown as any)._features;
+            const features: (StaticFeature | DynamicFeature<any>)[] = (client as unknown as any)._features;
             const minusCommands = features.filter(
                 (f) => (f as any).registrationType?.method != 'workspace/executeCommand'
             );
             minusCommands.push(new NerfedExecuteCommandFeature());
-            (languageClient as any)._features = minusCommands;
+            (client as any)._features = minusCommands;
 
             // Then start (which will cause the initialize request to be sent to pylance)
-            const languageClientDisposable = languageClient.start();
+            await client.start();
 
-            // After starting, wait for it to be ready
-            while (languageClient && !languageClient.initializeResult) {
-                await sleep(100);
-            }
-            if (languageClient) {
-                await languageClient.onReady();
-            }
+            // Expose client once it is fully initialized.
```

Comment:
I don't think you need this line anymore. Not used.

Reply:
I think languageClient is still exposed through middleware?

Reply:
No this variable isn't used. It's just assigned. You use 'client' instead everywhere, so you don't really need this variable.

Reply:
Oh didn't notice that as it wasn't in the changes. Yeah looks like it's needed for the lambda capture.